src/formatting.jl

```diff
@@ -258,11 +284,73 @@ function format_published_in(entry; include_date=true)
         ]
         str *= join(filter!(!isempty, parts), ", ")
     elseif entry.type == "unpublished"
-        parts = [get(entry.fields, "note", ""),]
-        str *= join(filter!(!isempty, parts), ", ")
+        if isempty(get(entry.fields, "note", ""))
+            @warn "unpublished $(entry.id) does not have a 'note'"
+        end
     end
     if include_date && !isempty(entry.date.year)
         str *= " ($(entry.date.year))"
     end
     return str
 end
+
+
+function format_note(entry)
+    return strip(get(entry.fields, "note", "")) |> tex2unicode
+end
+
+
+function format_eprint(entry)
+    eprint = entry.eprint.eprint
+    if isempty(eprint)
+        return ""
+    end
+    archive_prefix = entry.eprint.archive_prefix
+    primary_class = entry.eprint.primary_class
+    if isempty(archive_prefix) || (lowercase(archive_prefix) == "arxiv")
+        archive_prefix = "arXiv"
+    end
+    text = "$(archive_prefix):$eprint"
+    if !isempty(primary_class)
+        text *= " [$(primary_class)]"
+    end
+    link = ""
+    if archive_prefix == "arXiv"
+        link = "https://arxiv.org/abs/$eprint"
+    end
+    return linkify(text, link)
```

Comment:
Here we could add for `archive_prefix == "HAL"`
```suggestion
    if archive_preprint == "HAL"
        link = "https://hal.science/$(eprint)/"
    end
    if archive_preprint == "biorXiv"
        link = "https://www.biorxiv.org/content/$doi"
    end
    return linkify(text, link)
```

For HAL see e.g. https://hal.science/hal-03909427/, where the eprint id is `hal-XXXXXXXX`.

For biorXiv I am not sure they have IDs still, since e.g. https://www.biorxiv.org/content/10.1101/2023.03.06.531268v1 just uses the DOI as identifier – one could still do exactly that, use the doi, though that leaves the eprint field ignored in this case. I sketched the code above and hope that works as intended :)

Reply:
I'll add bioRxiv and HAL support in another PR after this is merged

Reply:
https://github.com/JuliaDocs/DocumenterCitations.jl/issues/35. Looks like the official ID for bioRxiv is only the last part of the DOI (excluding the `10.1101` prefix).

Reply:
A new PR is of course fine – and yes you are correct, since the first part of the DOI identifies the publisher.